src/ai/backend/agent/docker/agent.py

```diff
@@ -876,6 +881,28 @@ async def start_container(
                 },
             },
         }
+
+        seccomp_profile_pth = self.config_dir / "seccomp.json"
+
+        if seccomp_profile_pth.exists():
+            with open(seccomp_profile_pth, "r") as seccomp_file:
+                seccomp_profile = json.load(seccomp_file)
+
+                additional_syscalls = self.additional_syscalls
+
+                additional_syscall_rule = {
+                    "names": additional_syscalls,
+                    "action": "SCMP_ACT_ALLOW",
+                    "args": [],
+                    "comment": "Additionally allowed syscalls by Backend.AI Agent",
+                }
+
+                seccomp_profile["syscalls"].append(additional_syscall_rule)
```

Comment:
Wouldn't it be better to just update the copied seccomp.json file in-place and reference it further inside the `container_config`?

Reply:
The seccomp option can only be provided as an inline JSON string, and (unlike in the CLI) it does not seem to accept a file path.

So, we've decided to create and pass an inline JSON string here.

Issue Reference:
https://gitlab.com/gitlab-org/gitlab-runner/-/issues/27235